Update Day2.md name and formatting

superslalomracer opened the pull request:
Update the formatting of the Day2.md file to match with the current Lecture_template.md file, and various other formatting conventions discussed in issues #9 , #17, and #33. Change the header, add the date, declare code language. Also, change the file name to match the naming convention from Day2.md to Lecture_02.md. 

Note: There was an opened and closed PR with the same name. It was closed because it branched off of the template branch, which has not yet been accepted. This branch is branched off of main, which is currently even with the upstream main.

superslalomracer committed:
Update Day2.md formatting

Update the format of the Day2.md file to match the current formatting
convention as detailed in the Lecture_template.md file and described
in issue #9. Changes include adding a new header, the date of the
lecture, unifying the topic headers, and adding the language of each
code block.

superslalomracer committed:
Change Day2.md to Lecture_02.md

Change the name of the file for the day 2 notes, as fitting with the
current naming convention. No file content changed, only the file name.

kevinaangstadt commented:
@epuer18 could you please review this PR?  You may need to do this just in the comments due to permissions.

epuer18 commented:
The file name change is perfect and the commit message is concise and descriptive. Commit 9324349, namely Update Day.2md formatting was made before some of the adjustments to the template and finalizations on the style guide. Adding languages to code blocks and the adjustment of the headers looks great, but the titles could go back to how they were originally (I guess this is why multiple commits of small things are useful *sweaty smiley face). Commit message does look good and actually helped me localize changes better!

superslalomracer committed:
Update formatting to current guide

Update the general format of the lecture 2 file to match the current
template file. Specifically, the size of headers has changed to (##)
from (###).

superslalomracer commented:
I've just committed a few changes to fit to the current format.
